statServer/models/questions_model.ts

```diff
@@ -0,0 +1,73 @@
+/*
+El nuevo entrypoint tiene la siguiente estructura:
+
+GET /questions?since=2022-03-04
+
+{
+   "questionId": ".. ",  // Identificación de la pregunta 
+   "question": ".. ",  // Texto de la pregunta en español
+   "attempts":  #  // Número de veces que la pregunta se ha intentado 
+   "avgLatency": # // Tiempo promedio en segundo que la gente necesita para responder esta pregunta
+   "successProb": # //  La probabilidad de tener éxito respondiendo esta pregunta (valor entre cero y uno)
+}
+since: fecha para considerar los datos sólo después de la fecha dada, es opcional, si no está se consideran todos los datos disponibles.
+
+Para los textos de las preguntas se va a usar el resultado del issue Issue 54
+*/
+
+import { Questions } from '../types/questions';
+import {db} from "../db";
+import { RowDataPacket } from "mysql2";
+
+
+export const findAll = (callback: Function, date: any) => {
+const queryString = `
+
+SELECT
+    q.id AS questionId,
+    q.question,
+    COUNT(ua.question_id) AS attempts,
+    AVG(ua.latency / 1000) AS avgLatency,
+    SUM(CASE WHEN ua.attempts > 0 THEN 1 ELSE 0 END) / COUNT(ua.question_id) AS successProb
+FROM
+    question q
+LEFT JOIN
+    user_answer ua ON q.id = ua.question_id
+WHERE
+    CASE
+        WHEN EXISTS (
```

Comment:
Para la condición usemos un rango de fechas, si uno de las fechas falta la condición de ese extremo no se incluye . Si la fecha 'since' y 'to' están presente, entonces ua.answered >= since AND ua.answered < to, pero si está nulo since: ua.answered < to, y si ambas son nulas, retorna todo.

Reply:
Se realiza ajuste dentro de la lógica de programación se establece consulta por rango como se indica  la variable since y to, se organizan consultas de repuestas con orden de mayo a menor según el promedio de latencia, dejan como opcionales y se realizan pruebas desde postamn con URL = http://localhost:3300/album_stats/questions?since={since}&to={to} coon formato dd/mm/aaaa, si alguna en nula toma la información.

se realiza cambio den queryroutes y en question_models como indica el request.

`xport const getQuetionsByDates = (callback: Function, dates: any) => {

    let { since, to } = dates;
 
    if (since === '' || to === '') {
        console.log('cambio de since o to')
        since = '01/01/0001'
        to = '12/12/9999'
    }
    const queryString = `
    
    SELECT
        q.id AS questionId,
        q.question,
        COUNT(ua.question_id) AS attempts,
        AVG(ua.latency / 1000) AS avgLatency,
        SUM(CASE WHEN ua.attempts > 0 THEN 1 ELSE 0 END) / COUNT(ua.question_id) AS successProb
    FROM
        question q
    LEFT JOIN
        user_answer ua ON q.id = ua.question_id
    WHERE
        CASE
            WHEN EXISTS (
                SELECT 1
                FROM user_answer ua
                WHERE DATE(FROM_UNIXTIME(ua.answered_on/1000)) BETWEEN STR_TO_DATE("${since}", '%d/%m/%Y') AND STR_TO_DATE("${to}", '%d/%m/%Y') 
            )
            THEN DATE(FROM_UNIXTIME(ua.answered_on/1000)) BETWEEN STR_TO_DATE("${since}", '%d/%m/%Y') AND STR_TO_DATE("${to}", '%d/%m/%Y')
            
        END
    GROUP BY
        q.id, q.question
    ORDER BY  avgLatency DESC   
    ;
    ``

#55

Reply:
por favor revisar @edwin-navarrete